modules/social_features/social_topic/src/Plugin/GraphQL/SchemaExtension/TopicSchemaExtension.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace Drupal\social_topic\Plugin\GraphQL\SchemaExtension;
+
+use Drupal\graphql\GraphQL\ResolverBuilder;
+use Drupal\graphql\GraphQL\ResolverRegistryInterface;
+use Drupal\graphql\Plugin\GraphQL\SchemaExtension\SdlSchemaExtensionPluginBase;
+
+/**
+ * Adds topic data to the Open Social GraphQL API.
+ *
+ * @SchemaExtension(
+ *   id = "social_topic_schema_extension",
+ *   name = "Open Social - Topic Schema Extension",
+ *   description = "GraphQL schema extension for Open Social topic data.",
+ *   schema = "open_social"
+ * )
+ */
+class TopicSchemaExtension extends SdlSchemaExtensionPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function registerResolvers(ResolverRegistryInterface $registry) {
+    $builder = new ResolverBuilder();
+
+    $this->addQueryFields($registry, $builder);
+    $this->addTopicFields($registry, $builder);
+  }
+
+  /**
+   * Registers type and field resolvers in the shared registry.
+   *
+   * @param \Drupal\graphql\GraphQL\ResolverRegistryInterface $registry
+   *   The resolver registry.
+   * @param \Drupal\graphql\GraphQL\ResolverBuilder $builder
+   *   The resolver builder.
+   */
+  protected function addTopicFields(ResolverRegistryInterface $registry, ResolverBuilder $builder) {
+    $registry->addFieldResolver('Topic', 'id',
+      $builder->produce('entity_uuid')
+        ->map('entity', $builder->fromParent())
+    );
+
+    $registry->addFieldResolver('Topic', 'title',
+      $builder->compose(
+        $builder->produce('entity_label')
+          ->map('entity', $builder->fromParent()),
+        $builder->produce('uppercase')
+          ->map('string', $builder->fromParent())
```

Comment:
Why are we uppercasing the label here?

Reply:
I don't know, we did it everywhere if I'm not mistaken, but I'm also not sure if it is a good idea. Probably because of `composable-schemas.md` - html/modules/contrib/graphql/doc/advanced/composable-schemas.md

Reply:
Removed uppercase.